Plugins/SpatialGDK/SpatialGDKEditorToolbar/Source/SpatialGDKEditorToolbar/Private/InteropCodeGenerator/SpatialGDKEditorInteropCodeGenerator.cpp

```diff
@@ -127,36 +119,110 @@ FString GenerateIntermediateDirectory()
 	return AbsoluteCombinedIntermediatePath;
 }
 
-bool SpatialGDKGenerateInteropCode()
+void CreateSchemaDatabase(TArray<UClass*> Classes)
 {
-	const USpatialGDKEditorToolbarSettings* SpatialGDKToolbarSettings = GetDefault<USpatialGDKEditorToolbarSettings>();
-	TArray<UClass*> InteropGeneratedClasses;
+	AsyncTask(ENamedThreads::GameThread, [Classes]{
+		FString PackagePath = TEXT("/Game/Spatial/SchemaDatabase");
+
+		UPackage *Package = CreatePackage(nullptr, *PackagePath);
+
+		USchemaDatabase* SchemaDatabase = NewObject<USchemaDatabase>(Package, USchemaDatabase::StaticClass(), FName("SchemaDatabase"), EObjectFlags::RF_Public | EObjectFlags::RF_Standalone);
+
+		int ComponentId = 100010;
+		for (UClass* Class : Classes)
+		{
+			FSchemaData SchemaData;
+			SchemaData.SingleClientRepData = ComponentId++;
+			SchemaData.MultiClientRepData = ComponentId++;
+			SchemaData.HandoverData = ComponentId++;
+			SchemaData.ClientRPCs = ComponentId++;
+			SchemaData.ServerRPCs = ComponentId++;
+			SchemaData.CrossServerRPCs = ComponentId++;
+			SchemaData.NetMulticastRPCs = ComponentId++;
+
+			SchemaDatabase->ClassToSchema.Add(Class, SchemaData);
+		}
+
+		FAssetRegistryModule::AssetCreated(SchemaDatabase);
+		SchemaDatabase->MarkPackageDirty();
+
+		FString FilePath = FString::Printf(TEXT("%s%s"), *PackagePath, *FPackageName::GetAssetPackageExtension());
+		bool bSuccess = UPackage::SavePackage(Package, SchemaDatabase, EObjectFlags::RF_Public | EObjectFlags::RF_Standalone, *FPackageName::LongPackageNameToFilename(PackagePath, FPackageName::GetAssetPackageExtension()));
```

Comment:
We should have log this if it fails so that the user can know that this operation has failed